sw/device/silicon_creator/lib/drivers/alert.c

```diff
@@ -331,7 +331,7 @@ rom_error_t alert_config_check(lifecycle_state_t lc_state) {
           otp_read32(OTP_CTRL_PARAM_OWNER_SW_CFG_ROM_ALERT_DIGEST_RMA_OFFSET);
       break;
     default:
-      HARDENED_UNREACHABLE();
+      HARDENED_TRAP();
```

Comment:
Should this also have `OT_UNREACHABLE()`, since it's under `default`? I flagged these where I saw them, but I might just be missing something about where it's needed and where it isn't.
```suggestion
      HARDENED_TRAP();
      OT_UNREACHABLE();
```

Reply:
Thanks @jadephilipoom and apologies for not being clear. We need the `OT_UNREACHABLE()` macro only when `HARDENED_TRAP()` appears last in a  control path, e.g. this [line](https://github.com/lowRISC/opentitan/blob/80a893651c6936f2101d3fbc960cfc8da1a359ee/sw/device/silicon_creator/rom/sigverify_keys.c#L67) in `sigverify_keys.c`. Otherwise, build fails with:
```
sw/device/silicon_creator/rom_ext/sigverify_keys.c:152:1: error: non-void function does not return a value in all control paths [-Werror,-Wreturn-type]                                                             
}                                                                                                                                                                                                                   
^
```

Your comment made me go through my changes and I spotted one we didn't really need. You can't really under-use `OT_UNREACHABLE()` because compiler errors out. We don't want to overuse though because it enables the compiler to move our `unimp` blocks to the end of the function.